contracts/escrow-fee-distributor/src/contract.rs

```diff
@@ -0,0 +1,881 @@
+    }
+
+    if let Some(can_checkpoint_token) = can_checkpoint_token {
+        config.can_checkpoint_token = can_checkpoint_token;
+        attributes.push(Attribute::new(
+            "can_checkpoint_token",
+            can_checkpoint_token.to_string(),
+        ));
+    };
+
+    if let Some(max_limit_accounts_of_claim) = max_limit_accounts_of_claim {
+        config.max_limit_accounts_of_claim = max_limit_accounts_of_claim;
+        attributes.push(Attribute::new(
+            "max_limit_accounts_of_claim",
+            max_limit_accounts_of_claim.to_string(),
+        ));
+    };
+
+    CONFIG.save(deps.storage, &config)?;
+
+    Ok(Response::new().add_attributes(attributes))
+}
+
+/// ## Description
+/// Available the query messages of the contract.
+/// ## Queries
+/// * **QueryMsg::Config {}** Returns the base controls configs that contains in the [`Config`] object.
+///
+/// * **QueryMsg::AstroRecipientsPerWeek {}** Returns the list of accounts who will get ASTRO fees
+/// every week in the [`RecipientsPerWeekResponse`] object.
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn query(deps: Deps, env: Env, msg: QueryMsg) -> StdResult<Binary> {
+    match msg {
+        QueryMsg::FetchUserBalanceByTimestamp { user, timestamp } => {
+            Ok(to_binary(&query_user_balance(deps, env, user, timestamp)?)?)
+        }
+        QueryMsg::Config {} => Ok(to_binary(&query_config(deps)?)?),
+        QueryMsg::AstroRecipientsPerWeek {} => Ok(to_binary(&query_recipients_per_weeks(deps)?)?),
+        QueryMsg::VotingSupplyPerWeek {} => Ok(to_binary(&query_voting_supply_per_week(deps)?)?),
+        QueryMsg::FeeTokensPerWeek {} => Ok(to_binary(&query_tokens_per_week(deps)?)?),
+    }
+}
+
+fn query_voting_supply_per_week(deps: Deps) -> StdResult<Vec<Uint128>> {
+    let mut result: Vec<Uint128> = vec![];
+    for x in VOTING_SUPPLY_PER_WEEK.keys(deps.storage, None, None, Order::Ascending) {
+        let val = VOTING_SUPPLY_PER_WEEK.load(deps.storage, U64Key::from(x))?;
+        result.push(val);
+    }
+    Ok(result)
+}
+
+fn query_tokens_per_week(deps: Deps) -> StdResult<Vec<Uint128>> {
+    let mut result: Vec<Uint128> = vec![];
+    for x in TOKENS_PER_WEEK.keys(deps.storage, None, None, Order::Ascending) {
+        let val = TOKENS_PER_WEEK.load(deps.storage, U64Key::from(x))?;
+        result.push(val);
+    }
+    Ok(result)
+}
+
+/// ## Description
+/// Returns the user fee amount by the timestamp
+fn query_user_balance(deps: Deps, _env: Env, user: String, timestamp: u64) -> StdResult<Uint128> {
+    let config = CONFIG.load(deps.storage)?;
+    let user_voting_power: VotingPowerResponse = deps.querier.query_wasm_smart(
+        &config.voting_escrow,
+        &VotingQueryMsg::UserVotingPowerAt {
+            user,
+            time: timestamp,
+        },
+    )?;
+
+    let mut user_fee_balance = Uint128::zero();
+    let current_period = get_period(timestamp);
+
+    if !user_voting_power.voting_power.is_zero() {
```

Comment:
This code copypasted from calc_claim_amount(). Maybe we should combine it into one function?